### PR TITLE
DR-741 skaffold helm

### PR DESCRIPTION
**Pull request by thathert**

### CODE REVIEW

**Review by marikomedlock — APPROVED** [review]
I have two questions about how this works. Apologies if they're basic, I'm still wrapping my head around how our new multi-tenant project works.:
  1. Why does it use the ms namespace instead? Is this something we'll need to change locally to push images for our own projects?
  2. The PIZZA is just a nice tag name to search on when we're looking for the image name to update our individual namespaces, right?

**Comment by smark88** [comment]
> I have two questions about how this works. Apologies if they're basic, I'm still wrapping my head around how our new multi-tenant project works.:
> 
> 1. Why does it use the ms namespace instead? Is this something we'll need to change locally to push images for our own projects?
> 2. The PIZZA is just a nice tag name to search on when we're looking for the image name to update our individual namespaces, right?

This was a test deploy to my namespace specifically in the future maybe the skaffold -f flag may need to be used. 

Pizza was to see if we could tag the image with a specific name and deploy it to a namespace.

**Comment by marikomedlock** [comment]
> > I have two questions about how this works. Apologies if they're basic, I'm still wrapping my head around how our new multi-tenant project works.:
> > 
> > 1. Why does it use the ms namespace instead? Is this something we'll need to change locally to push images for our own projects?
> > 2. The PIZZA is just a nice tag name to search on when we're looking for the image name to update our individual namespaces, right?
> 
> This was a test deploy to my namespace specifically in the future maybe the skaffold -f flag may need to be used.
> 
> Pizza was to see if we could tag the image with a specific name and deploy it to a namespace.

Got it, thank you for the explanation.

**Review by fboulnois — COMMENTED** [review]
What does the `jib` arg to `jib` do?

**Comment by smark88** [comment]
@thathert this should be ready now. Folks can have their own local skaffold.yaml by running the command in the readme.

File: README.md [hunk 1]
```diff
@@ -145,3 +145,14 @@ in a directory other than a git cloned workspace, run:
 `swagger-codegen generate -i path/to/data-repository-openapi.yaml -l spring -c path/to/config.json`
 
 Then copy the files you want into the source tree
+
+## skaffold
+To render your own local skffold.yaml run the following with your initials
```
Comment (fboulnois) [inline README.md]: skaffold.yaml

Comment (fboulnois) [inline README.md:150]:
```suggestion
To render your own local skaffold.yaml run the following with your initials
```

File: skaffold.yaml.template [hunk 2]
```diff
@@ -0,0 +1,34 @@
+apiVersion: skaffold/v2alpha2
+kind: Config
+build:
+  tagPolicy:
+    envTemplate:
+      template: "{{.IMAGE_NAME}}:{{.IMAGE_TAG}}"
```
Comment (thathert) [inline skaffold.yaml.template]: I think there is a way to do this without needing to get the has into an env variable. Something like this:

```
tagPolicy:
    gitCommit: {}
```
From the gitCommit tagger: https://skaffold.dev/docs/pipeline-stages/taggers/

Comment (smark88) [reply skaffold.yaml.template]: I know, but i couldn't figure out how to change the tag in the helm chart, if you wanna take a stable at it that'd be great.

File: skaffold.yaml.template [hunk 3]
```diff
@@ -15,6 +15,7 @@ deploy:
     flags:
       upgrade:
         - --install
+        - --debug
```
Comment (thathert) [inline skaffold.yaml.template:18]: Did you mean to leave this in?

Comment (smark88) [reply skaffold.yaml.template:18]: yes it spits out what is actually being deployed (raw kubernetes yamls), I think it is helpful see snippet below
```
---
# Source: datarepo/charts/oidc-proxy/templates/oidc-ingress.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: ms-jade-oidc-proxy
  labels:
    helm.sh/chart: oidc-proxy-0.0.3
    app.kubernetes.io/name: oidc-proxy
    app.kubernetes.io/instance: ms-jade
    app.kubernetes.io/managed-by: Helm
  annotations:
    kubernetes.io/ingress.global-static-ip-name: "jade-dev-ms"
    kubernetes.io/ingress.allow-http: "false"
spec:
  tls:
    - secretName: oidc-tls-key
  rules:
  - http:
      paths:
      - path: /*
        backend:
          serviceName: ms-jade-oidc-proxy
          servicePort: http
```

**Review by thathert — COMMENTED** [review]
I can't approve this because I technically made it, but this is really @smark88 's now. LGTM

**Review by smark88 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (fboulnois) [reply README.md:150]: This seems to still be in: https://github.com/DataBiosphere/jade-data-repo/blame/develop/README.md#L150

**Pull request merged**